src/halmos/__main__.py

```diff
@@ -281,18 +276,15 @@ def rendered_trace(context: CallContext) -> str:
         return output.getvalue()
 
 
-def rendered_calldata(calldata: ByteVec) -> str:
-    return hexify(calldata.unwrap()) if calldata else "0x"
+def rendered_calldata(calldata: ByteVec, contract_name: str = None) -> str:
+    return hexify(calldata.unwrap(), contract_name) if calldata else "0x"
 
 
 def render_trace(context: CallContext, file=sys.stdout) -> None:
-    # TODO: label for known addresses
-    # TODO: decode calldata
-    # TODO: decode logs
-
     message = context.message
     addr = unbox_int(message.target)
     addr_str = str(addr) if is_bv(addr) else hex(addr)
+    addr_str = DeployAddressMapper().get_deployed_contract(addr_str)
```

Comment:
For converting from address to contract name

Reply:
```suggestion
    # check if we have a contract name for this address in our deployment mapper
    addr_str = DeployAddressMapper().get_deployed_contract(addr_str)
```

Reply:
Quick test to see how foundry deals with the same contract deployed at multiple addresses:

This:

```solidity
contract CounterTest is Test {
    function setUp() public {
        Counter counter = new Counter();
        counter.setNumber(0);

        Counter counter1 = new Counter();
        counter1.setNumber(1);

        Counter counter2 = new Counter();
        counter2.setNumber(2);
    }
}
```

Is rendered like this:

```
[PASS] testIncrement() (gas: 28334)
Traces:
  [315303] CounterTest::setUp()
    ├─ [49499] → new Counter@0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f
    │   └─ ← [Return] 247 bytes of code
    ├─ [2390] Counter::setNumber(0)
    │   └─ ← [Stop] 
    ├─ [49499] → new Counter@0x2e234DAe75C793f67A35089C9d99245E1C58470b
    │   └─ ← [Return] 247 bytes of code
    ├─ [22290] Counter::setNumber(1)
    │   └─ ← [Stop] 
    ├─ [49499] → new Counter@0xF62849F9A0B5Bf2913b396098F7c7019b51A820a
    │   └─ ← [Return] 247 bytes of code
    ├─ [22290] Counter::setNumber(2)
    │   └─ ← [Stop] 
    └─ ← [Stop] 
```

I don't love the ambiguity about which Counter instance is called, but we're on par with foundry so fine to keep it as is for now

Reply:
@karmacoma-eth 
Thank you for your feedback!
I hadn't considered this example. Now I understand the concern about the ambiguity regarding which Counter instance is called.